### PR TITLE
Ros control tutorials

### PR DESCRIPTION
**Pull request by paulbovbel**
Address #12 

Primary change to early tutorials is to clean-up formatting, and rotate robot to be x-forward (it was previously y-forward).

Will implement the following updates to wiki:
 - Minor updates to material tags and gazebo materials http://wiki.ros.org/urdf/Tutorials/Building%20a%20Visual%20Robot%20Model%20with%20URDF%20from%20Scratch - 

 - Minor updates for dynamic parameters/gazebo dynamic tags http://wiki.ros.org/urdf/Tutorials/Adding%20Physical%20and%20Collision%20Properties%20to%20a%20URDF%20Model

- Major update with xacro cleanup
http://wiki.ros.org/urdf/Tutorials/Using%20Xacro%20to%20Clean%20Up%20a%20URDF%20File

- New section required: Integrating with Gazebo and ROS Control

### CODE REVIEW

**Comment by paulbovbel** [comment]
To see a demo, in a ROS workspace:

git clone git@github.com:clearpathrobotics/urdf_tutorial.git -b ros-control-tutorials
rosdep install --from-paths . --ignore-src -y

roslaunch urdf_tutorial control.launch model:=urdf_tutorial/urdf/r2d2.xacro 
rosrun rqt_robot_steering rqt_robot_steering
# set topic to /r2d2_diff_drive_controller/cmd_vel

**Comment by DLu** [comment]
Overall, looks good, and thanks for doing this. 

However, its missing dependencies on controller_manager and gazebo_ros, as well as gazebo_worlds which seems to be deprecated. At the moment, this is what's stopping me from running the demo.

**Comment by paulbovbel** [comment]
@DLu, thanks for the attention, I've added a roslaunch check and resolved all dependency issues, lemme know if there's any other trouble.

**Comment by DLu** [comment]
This is the first part of the tutorial that really splits along distributions, since position_controllers/JointGroupPositionController doesn't exist prior to indigo. I've created new tags to reflect this. 

However, since hydro won't get any new versions, I think we don't have change the PR.

File: launch/display.launch [hunk 1]
```diff
@@ -1,9 +1,14 @@
 <launch>
-	<arg name="model" />
-	<arg name="gui" default="False" />
-	<param name="robot_description" textfile="$(arg model)" />
-	<param name="use_gui" value="$(arg gui)"/>
-	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
-	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
-	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find urdf_tutorial)/urdf.rviz" required="true" />
+
+	<arg name="model" default="$(find urdf_tutorial)/urdf/r2d2.xacro"/>
```
Comment (DLu) [inline launch/display.launch]: One weird tab.

File: launch/control.launch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  <!-- This node loads the two controllers into a controller manager (real or simulated). The
+  controllers are defined in config/control.yaml -->
+  <node name="r2d2_controller_spawner" pkg="controller_manager" type="spawner"
+    args="r2d2_joint_state_controller
+          r2d2_diff_drive_controller
+          r2d2_head_controller
+          r2d2_gripper_controller
+          --shutdown-timeout 3"/>
+
```
Comment (DLu) [inline launch/control.launch]: Any reason to not include the rqt_robot_steering in here (with topic remapping)?

**Comment by DLu** [comment]
I ran the tutorial and everything worked out great. Nice work. Turns out I made a not very stable r2d2 model; I kept knocking him over. 

In addition to the two minor things above, one future request would be to add the textures, but that's not needed for this control work.

**Comment by paulbovbel** [comment]
Done. Intertial info should definitely be looked over at some point (open ticket?), but the model so far is fine for demo/tutorial purposes.

**Comment by DLu** [comment]
Merged. I started #15 to track the new wiki changes.

**Comment by DLu** [comment]
Thanks again for this valuable contribution.

**Pull request merged**